packages/ui/react-ui-thread/src/types.ts

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import type { AvatarStatus } from '@dxos/react-ui';
+
+export type MessageMetadata = {
+  id: string;
+  authorId: string;
+  authorName?: string;
+  authorImgSrc?: string;
+  authorStatus?: AvatarStatus;
+};
+
+export type MessageEntityBlock = {
```

Comment:
we have to use the existing reactive types here
i think it's likely premature to factor threads out into a separate package until we complete the schema work.

Reply:
I’ve followed Table’s precedent here, making this into a generic that consumers can enforce their type-safety with.